cypress/utils/commands.js

```diff
@@ -13,3 +28,244 @@ Cypress.Commands.add('getElementsByTestIds', (testIds, options = {}) => {
+  });
+});
+
+Cypress.Commands.add('importSavedObjects', (fixturePath, overwrite = true) => {
+  const sendImportRequest = (ndjson) => {
+    const url = `${Cypress.config().baseUrl}/api/saved_objects/_import?${
+      overwrite ? `overwrite=true` : ''
+    }`;
+
+    const formData = new FormData();
+    formData.append('file', ndjson, 'savedObject.ndjson');
+
+    cy.log('importSavedObject')
+      .request({
+        method: 'POST',
+        url,
+        headers: {
+          'content-type': 'multipart/form-data',
+          'osd-xsrf': true,
+        },
+        body: formData,
+      })
+      .then((response) => {
+        if (response.body.errors) {
+          console.error(response.body.items);
+          throw new Error('Import failed');
+        }
+      });
+  };
+
+  cy.fixture(fixturePath)
+    .then((file) => Cypress.Blob.binaryStringToBlob(file))
+    .then((ndjson) => {
+      sendImportRequest(ndjson);
+    });
+});
+
+Cypress.Commands.add('deleteSavedObject', (type, id, options = {}) => {
+  const url = `${Cypress.config().baseUrl}/api/saved_objects/${type}/${id}`;
+
+  return cy.request({
+    method: 'DELETE',
+    url,
+    headers: {
+      'osd-xsrf': true,
+    },
+    failOnStatusCode: false,
+    ...options,
+  });
+});
+
+Cypress.Commands.add('deleteSavedObjectByType', (type, search) => {
+  const searchParams = new URLSearchParams({
+    fields: 'id',
+    type,
+  });
+
+  if (search) {
+    searchParams.set('search', search);
+  }
+
+  const url = `${
+    Cypress.config().baseUrl
+  }/api/opensearch-dashboards/management/saved_objects/_find?${searchParams.toString()}`;
+
+  return cy.request(url).then((response) => {
+    console.log('response', response);
+    response.body.saved_objects.map(({ type, id }) => {
+      cy.deleteSavedObject(type, id);
+    });
+  });
+});
+
+Cypress.Commands.add('createIndexPattern', (id, attributes, header = {}) => {
+  const url = `${Cypress.config().baseUrl}/api/saved_objects/index-pattern/${id}`;
+
+  cy.request({
+    method: 'POST',
+    url,
+    headers: {
+      'content-type': 'application/json;charset=UTF-8',
+      'osd-xsrf': true,
+      ...header,
+    },
+    body: JSON.stringify({
+      attributes,
+      references: [],
+    }),
+  });
+});
+
+Cypress.Commands.add('createDashboard', (attributes = {}, headers = {}) => {
+  const url = `${Cypress.config().baseUrl}/api/saved_objects/dashboard`;
+
+  cy.request({
+    method: 'POST',
+    url,
+    headers: {
+      'content-type': 'application/json;charset=UTF-8',
+      'osd-xsrf': true,
+      ...headers,
+    },
+    body: JSON.stringify({
+      attributes,
+    }),
+  });
+});
+
+Cypress.Commands.add('changeDefaultTenant', (attributes, header = {}) => {
+  const url = Cypress.env('openSearchUrl') + '/_plugins/_security/api/tenancy/config';
+
+  cy.request({
+    method: 'PUT',
+    url,
+    headers: {
+      'content-type': 'application/json;charset=UTF-8',
```

Comment:
we can modify the cy.request which i think the security plugin commands are doing to just add some of these copy and paste request headers